app/controllers/ChannelTestsController.scala

```diff
@@ -151,4 +156,130 @@ abstract class ChannelTestsController[T <: ChannelTest[T] : Decoder : Encoder](
     }
   }
 
+  /**
+    * Handlers for test list ordering
+    */
+
+  def lockList = authAction.async { request =>
+    runWithLockStatus { case VersionedS3Data(lockStatus, lockFileVersion) =>
+      logger.info(s"User ${request.user.email} is locking $channel test list")
+
+      if (!lockStatus.locked) {
+        val newLockStatus = LockStatus.locked(request.user.email)
+
+        setLockStatus(VersionedS3Data(newLockStatus, lockFileVersion)).map(_ => Ok("locked"))
+      } else {
+        logger.info(s"User ${request.user.email} failed to take control of $channel test list because it was already locked")
+        IO.succeed(Conflict(s"File $channel is already locked"))
+      }
+    }
+  }
+
+  def unlockList = authAction.async { request =>
+    runWithLockStatus { case VersionedS3Data(lockStatus, lockFileVersion) =>
+      logger.info(s"User ${request.user.email} is unlocking $channel test list")
+
+      if (lockStatus.email.contains(request.user.email)) {
+        setLockStatus(VersionedS3Data(LockStatus.unlocked, lockFileVersion)).map(_ => Ok("unlocked"))
+      } else {
+        logger.info(s"User ${request.user.email} tried to unlock $channel test list, but they did not have a lock")
+        IO.succeed(BadRequest(s"$channel test list is not currently locked by this user"))
+      }
+    }
+  }
+
+  def takeControlOfList = authAction.async { request =>
+    runWithLockStatus { case VersionedS3Data(lockStatus, lockFileVersion) =>
+      logger.info(s"User ${request.user.email} is force-unlocking $channel test list, taking it from ${lockStatus.email}")
+
+      setLockStatus(VersionedS3Data(LockStatus.locked(request.user.email), lockFileVersion)).map(_ => Ok("unlocked"))
+    }
+  }
+
+  def reorderList = authAction.async(circe.json[List[String]]) { request =>
+    runWithLockStatus { case VersionedS3Data(lockStatus, lockFileVersion) =>
+      if (lockStatus.email.contains(request.user.email)) {
+        logger.info(s"${request.user.email} is reordering $channel list")
+
+        val testNames: List[String] = request.body
+        val result = for {
+          _ <- dynamo.setPriorities(testNames, channel)
+          _ <- setLockStatus(VersionedS3Data(LockStatus.unlocked, lockFileVersion))
+        } yield ()
+
+        result
+          .map(_ => Ok("updated"))
+          .mapError { error =>
```

Comment:
I think is there some kind of tapError thing in ZIO?  so it lets you log the error "on the way past" effectively